test/benchmarks/squad_to_dpr.py

```diff
@@ -0,0 +1,320 @@
+    document_store: BaseDocumentStore = store_factory.get_document_store()
+
+    # 3. Load data into the document store
+    document_store.add_eval_data(squad_impossible_path.as_posix(), doc_index="document",
+                                 preprocessor=preprocessor)
+
+    # 4. Prepare retriever
+    retriever_factory = HaystackRetriever(document_store=document_store,
+                                          retriever_type=retriever_type_config[0],
+                                          **retriever_type_config[1])
+    retriever = retriever_factory.get_retriever()
+
+    # 5. Get embeddings if needed
+    if retriever_type_config[0] in ["DensePassageRetriever", "EmbeddingRetriever"]:
+        document_store.update_embeddings(retriever)
+
+    # 6. Find positive and negative contexts and create new dataset
+    iter_DPR = create_dpr_training_dataset(squad_data=squad_data,
+                                           retriever=retriever,
+                                           num_hard_negative_ctxs=num_hard_negative_ctxs)
+
+    # 7. Split (train, dev, test) and save dataset
+    total_nb_questions = get_number_of_questions(squad_data)
+    split_and_save_dataset(iter_dpr=iter_DPR,
+                           dpr_output_path=dpr_output_path,
+                           total_nb_questions=total_nb_questions)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description="Convert a SQuAD JSON format dataset to DPR format.")
+    parser.add_argument("--squad_file_path", dest="squad_file_path",
+                        help="A dataset with a SQuAD JSON format.",
+                        metavar="SQUAD_in", required=True)
+    parser.add_argument("--dpr_output_path", dest="dpr_output_path",
+                        help="The name of the DPR JSON formatted output file",
+                        metavar="DPR_out", required=True)
+    parser.add_argument("--num_hard_negative_ctxs", dest="nb_hard_neg_ctxs",
+                        help="Number of hard negative contexts to use",
+                        metavar="num_hard_negative_ctxs",
+                        default=30)
+    args = parser.parse_args()
+
+    preprocessor = PreProcessor(split_length=100, split_overlap=0, clean_empty_lines=False,
+                                clean_whitespace=False)
+    squad_file_path = Path(args.squad_file_path)
+    dpr_output_path = Path(args.dpr_output_path)
+    num_hard_negative_ctxs = args.num_hard_negative_ctxs
+
+    retriever_dpr_config = {
+        "use_gpu": True,
+    }
+    store_dpr_config = {
+        "embedding_field": "embedding",
+        "embedding_dim": 768,
+    }
+
+    retriever_bm25_config = {}
+
+    main(squad_file_path=squad_file_path, dpr_output_path=dpr_output_path,
+         document_store_type_config=("ElasticsearchDocumentStore", store_dpr_config),
+         # retriever_type_config=("ElasticsearchRetriever", retriever_bm25_config), # dpr
```

Comment:
Having the commented out alternative retriever_type_config is a great idea! Makes for easy switching between DPR and BM25. I assume you actually meant to have
`retriever_type_config=("DensePassageRetriever", retriever_dpr_config), # dpr`
or something similar?

Reply:
yes, it is another typo. Should read `retriever_type_config=("DensePassageRetriever", retriever_dpr_config), # dpr` indeed.